### PR TITLE
Add JAVA_OPTS_APPEND option from legacy to X (resolves #9188)

### LINKED ISSUE #9188: Provide an option like `JAVA_OPTS_APPEND` also for Quarkus based Keycloak.X container.
### Description

For Keycloak.X it's currently not possible to add custom Java properties on startup by just providing the ones you want to set without overriding the complete default `JAVA_OPTS` set in `kc.sh|.bat`. This makes it hard to adjust the environment to custom needs without modifying the `kc.sh` script, which makes updates more complex.
I therefore love to see a possibility to set additional `JAVA_OPTS` without overriding the defaults.

### Discussion

https://github.com/keycloak/keycloak/discussions/8654#discussioncomment-1793426

### Motivation

In the legacy, Wildfly-based Keycloak Docker container, there is an environment variable named `JAVA_OPTS_APPEND`, which allows us to append various Java options to the default configured ones.

This is currently missing for Keycloak.X.
Using `JAVA_TOOL_OPTIONS` does not work for overriding the default set `JAVA_OPTS` in the `kc.sh|bat` script, e.g. it is currently not possible to set custom `-Xmx` and `-Xms` properties.

### Details

_No response_

### PR DESCRIPTION
**Pull request by dasniko**
This adds the option to add additional, custom JAVA_OPTS to the server startup scripts, without the need to overwrite the complete defaults.

resolves #9188

### CODE REVIEW

File: distribution/server-x-dist/src/main/content/bin/kc.sh [hunk 1]
```diff
@@ -68,6 +68,11 @@ else
    echo "JAVA_OPTS already set in environment; overriding default settings with values: $JAVA_OPTS"
 fi
 
+if [ "x$JAVA_OPTS_APPEND" != "x" ]; then
+  echo "Appending addional Java properties to JAVA_OPTS: $JAVA_OPTS_APPEND"
```
Comment (DGuhr) [inline distribution/server-x-dist/src/main/content/bin/kc.sh:72]: typo: "additional"

Comment (DGuhr) [inline distribution/server-x-dist/src/main/content/bin/kc.sh:72]: typo: "additional"

Comment (DGuhr) [inline distribution/server-x-dist/src/main/content/bin/kc.sh:72]: typo: "additional"

**Review by DGuhr — CHANGES_REQUESTED** [review]
Apart from the typos, lgtm. Thanks for the contribution! Would be great if you could fix the typos, so we get this in :)

File: distribution/server-x-dist/src/main/content/bin/kc.sh (hunk 1 above)
Comment (dasniko) [reply distribution/server-x-dist/src/main/content/bin/kc.sh]: done.

File: distribution/server-x-dist/src/main/content/bin/kc.bat [hunk 2]
```diff
@@ -66,6 +66,11 @@ if not "x%JAVA_OPTS%" == "x" (
   set "JAVA_OPTS=-Xms64m -Xmx512m -XX:MetaspaceSize=96M -XX:MaxMetaspaceSize=256m -Djava.net.preferIPv4Stack=true"
 )
 
+if not "x%JAVA_OPTS_APPEND%" == "x" (
+  echo "Appending addional Java properties to JAVA_OPTS: %JAVA_OPTS_APPEND%"
```
[the remark this answers is not in the document]
Comment (dasniko) [reply distribution/server-x-dist/src/main/content/bin/kc.bat]: done

**Review by DGuhr — APPROVED** [review]
LGTM now, thanks!

**Review by pedroigor — APPROVED** [review]

**Review by stianst — APPROVED** [review]

**Pull request merged**